### PR TITLE
Add Planner E2E tests

### PR DESCRIPTION
**Pull request by vanwagonet**
[run-nightly]
refs: MBL-14015
affects: none
release note: none

This isn't meant to represent the full set of tests we want on the screen, just a start.

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/9uy7rj6) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F9uy7rj6)
teacher: [Link](https://rebrand.ly/vvizwj9) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fvvizwj9)
parent: [Link](https://rebrand.ly/3mjhnmy) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F3mjhnmy)

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1023980&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-14015](https://instructure.atlassian.net/browse/MBL-14015)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.55% | 93.55% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Parent/ParentE2ETests/PlannerTests.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            PlannerCalendar.monthButton.tap()
+        }
+        var shown = Date()
+        while calendar.compare(shown, to: reference, toGranularity: .month) != .orderedSame {
+            let isPast = calendar.compare(shown, to: reference, toGranularity: .month) == .orderedAscending
+            let halfScreen = UIScreen.main.bounds.width / 2
+            let daysCenter = PlannerCalendar.monthButton.relativeCoordinate(x: 0, y: 1)
+                .withOffset(CGVector(dx: halfScreen, dy: 50))
+            let dragTo = daysCenter.withOffset(CGVector(dx: isPast ? -halfScreen : halfScreen, dy: 0))
+            daysCenter.press(forDuration: 0, thenDragTo: dragTo)
```
Comment (cobbal) [inline Parent/ParentE2ETests/PlannerTests.swift]: Works, but could be easier

```suggestion
            let element = PlannerCalendar.dayButton(for: shown).waitToExist()
            if isPast {
                element.rawElement.swipeLeft()
            } else {
                element.rawElement.swipeRight()
            }
```

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

File: Parent/ParentE2ETests/PlannerTests.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                PlannerCalendar.dayButton(for: shown).swipeRight()
+            }
+            shown = shown.addMonths(isPast ? 1 : -1)
+        }
+        PlannerCalendar.dayButton(for: reference).tap()
+    }
+
+    func testPlanner() {
+        PlannerList.event(id: "2233").waitToExist()
+        XCTAssert(PlannerCalendar.dayButton(year: y, month: m, day: 1).label().contains("1, \(y), 1 event"))
```
Comment (vanwagonet) [inline Parent/ParentE2ETests/PlannerTests.swift]: I can't wait for the dots to exist to know the data has loaded, so I use the list to know the data has loaded, then check the dots for each day after I check the list.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024136&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024137&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Parent/ParentE2ETests/PlannerTests.swift (hunk 2 above)
Comment (cobbal) [reply Parent/ParentE2ETests/PlannerTests.swift]: The function `waitUntil` in `Element.swift` might be an easier way to solve this. It's the sledgehammer I tend to reach for when I want to be sure that a label eventually gets the right value

Comment (vanwagonet) [reply Parent/ParentE2ETests/PlannerTests.swift]: Good to have more tools. I think I'm ok with just intermixing the list & dots assertions for this though. If you disagree I can change it.

Comment (cobbal) [reply Parent/ParentE2ETests/PlannerTests.swift]: Yeah, this way seems fine to me too. I'll approve once bitrise does

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024138&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024165&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**